### DISCUSSION TITLE
How can a client efficiently update a large number of cells in a SharedMatrix?

### DISCUSSION THREAD

**Discussion by jogoertzen-stantec** [microsoft/FluidFramework#11935] [General]
_Note: I posted this same question [here](https://stackoverflow.com/questions/72106213/how-can-a-client-efficiently-update-a-large-number-of-cells-in-a-sharedmatrix) a few months ago, but I think this may be a better place to ask it._

I want to update the values of a large number (eg. 10,000) of cells in a SharedMatrix via setCells, but I don't want other connected clients to get 10,000 cellsChanged events (ie. one event for each cell).

Is it possible to use something like orderSequentially or groupOperation to somehow batch all the updates together so that it results in a single operation being broadcast to other connected clients?

I am using the AzureClient with Angular and connecting to a tinylicious server.

**Comment by DLehenbauer** [comment]
You will always get separate change notifications, even when the set operations are grouped.

Most customers coalesce events by deferring action until after the runtime has drained the queue of pending ops, typically by using `requestAnimationFrame` or similar.

I recommend you have a look at the introduction to #4048, which describes the atomicity guarantees you get by deferring the app's response to change events.  Note that `orderSequentially()` is no longer necessary now that the default FlushMode has changed.
